File: 102. Binary Tree Level Order Traversal/solution.js
var levelOrder = function(root) {
    let levels = [];
    
    if (isNil(root)) {
       return levels; 
    }

    let thisLevel = [root];

    while(true) {
        let nextLevel = [];
        let level = [];

        for (const node of thisLevel) {
            if (!isNil(node.left)) {
             nextLevel.push(node.left);
            }
    
            if (!isNil(node.right)) {
             nextLevel.push(node.right);
            }
    
            level.push(node.val);
        }

        levels.push(level);

        if (nextLevel.length === 0) {
            return levels
        } else {
            thisLevel = nextLevel;
        }
    }
}

File: 102. Binary Tree Level Order Traversal/solution.go
package main

func levelOrder(root *TreeNode) [][]int {
	var levels [][]int

	if root == nil {
		return levels
	}

	thisLevel := []*TreeNode{root}

	for {
		var nextLevel []*TreeNode
		var level []int
		for _, node := range thisLevel {
			if node.Left != nil {
				nextLevel = append(nextLevel, node.Left)
			}

			if node.Right != nil {
				nextLevel = append(nextLevel, node.Right)
			}

			level = append(level, node.Val)
		}

		levels = append(levels, level)

		if len(nextLevel) == 0 {
			return levels
		} else {
			thisLevel = nextLevel
		}
	}
}
